src/potdata/sampler.py

```diff
@@ -175,10 +182,71 @@ def __init__(
         self.reachable_ratio = reachable_ratio
         self.core_ratio = core_ratio
 
-        self._indices: list[int] = None
+        self._indices: list[int] = []
 
         np.random.seed(seed)
 
+    def _get_soap_vectors(self, data: list[Structure]) -> list[np.ndarray]:
```

Comment:
This function corresponds to step 1 above. 

- You should only create all the soap vectors for each `Structure`, not do any subselection, that is for step 2.
- The input is a list of pymatgen`Structure`. you can use `ASEAtomsAdaptor` to convert it to ASE Atoms. Then you can provide it to  `soap_desc.create`
- You should not do any hard code. stuff like `traj[-6000::5]` are absolute hard coding. This function should be general. We will use other sampler to select the steps. So, here you covert all the steps into soap vectors.

Reply:
1. Fixed, separate the original codes to step 1 and step 2 individually
2. Fixed
3. Fixed